nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/kafka/KafkaUtils.java

```diff
@@ -189,35 +218,52 @@ public boolean topicExists(String topic) {
         return JavaConversions.seqAsJavaList(topics);
     }
 
-    public List<String> listConsumers(Properties props, String topic) {
-        return listAllConsumers(props).stream()
-                                      .filter(consumer -> getConsumerTopics(props, consumer).contains(topic))
-                                      .collect(Collectors.toList());
+    public List<String> listConsumers(String topic) {
+        return listAllConsumers().stream().filter(consumer -> getConsumerTopics(consumer).contains(topic)).collect(
+                Collectors.toList());
     }
 
-    protected List<String> getConsumerTopics(Properties props, String group) {
-        AdminClient client = AdminClient.create(props);
-        return JavaConversions.mapAsJavaMap(client.listGroupOffsets(group))
+    protected List<String> getConsumerTopics(String group) {
+        return JavaConversions.mapAsJavaMap(getAdminClient().listGroupOffsets(group))
                               .keySet()
                               .stream()
                               .map(TopicPartition::topic)
                               .collect(Collectors.toList());
     }
 
-    public List<String> listAllConsumers(Properties props) {
-        List<String> ret = new ArrayList<>();
-        AdminClient client = AdminClient.create(props);
-        // this returns only consumer group that use the subscribe API (known by coordinator)
-        scala.collection.immutable.List<GroupOverview> groups = client.listAllConsumerGroupsFlattened();
-        Iterator<GroupOverview> iter = groups.iterator();
-        GroupOverview group;
-        while (iter.hasNext()) {
-            group = iter.next();
-            if (group != null) {
-                ret.add(group.groupId());
+    protected org.apache.kafka.clients.admin.AdminClient getNewAdminClient() {
+        if (newAdminClient == null) {
+            newAdminClient = org.apache.kafka.clients.admin.AdminClient.create(adminProperties);
+        }
+        return newAdminClient;
+    }
+
+    protected AdminClient getAdminClient() {
+        if (adminClient == null) {
+            adminClient = AdminClient.create(adminProperties);
+        }
+        return adminClient;
+    }
+
+    public List<String> listAllConsumers() {
+        long now = System.currentTimeMillis();
+        if (allConsumers == null || (now - allConsumersTime) > ALL_CONSUMERS_CACHE_TIMEOUT_MS) {
+            allConsumers = new ArrayList<>();
+            // this returns only consumer group that use the subscribe API (known by coordinator)
+            scala.collection.immutable.List<GroupOverview> groups = getAdminClient().listAllConsumerGroupsFlattened();
+            Iterator<GroupOverview> iterator = groups.iterator();
+            GroupOverview group;
+            while (iterator.hasNext()) {
+                group = iterator.next();
```

Comment:
You can also use a `for` look like this to make it a bit clearer it's a basic loop:

    for (Iterator<GroupOverview> iterator = groups.iterator(); iterator.hasNext(); ) {
        GroupOverview group = iterator.next();